matrix-sdk/src/main/java/org/matrix/androidsdk/MXDataHandler.java

```diff
@@ -1025,24 +1021,20 @@ private void manageAccountData(Map<String, Object> accountData, boolean isInitia
     /**
      * Refresh the push rules from the account data events list
      *
-     * @param events the account data events.
+     * @param accountDataElements the account data events list
      */
-    private void managePushRulesUpdate(List<Map<String, Object>> events) {
-        for (Map<String, Object> event : events) {
-            String type = (String) event.get("type");
-
-            if (TextUtils.equals(type, "m.push_rules")) {
-                if (event.containsKey("content")) {
-                    Gson gson = JsonUtils.getGson(false);
+    private void managePushRulesUpdate(List<AccountDataElement> accountDataElements) {
+        for (AccountDataElement accountDataElement : accountDataElements) {
```

Comment:
I think type check should be done on the top method `manageAccountData` as we repeat for loops at the moment.

Reply:
done